packages/transformer/src/test/TestUtils/IModelTestUtils.ts

```diff
@@ -1117,6 +1130,68 @@ export class IModelTestUtils {
     );
   }
 
+  public static findAndAssertTargetScopeProvenance(
+    master: TimelineIModelState,
+    branch: TimelineIModelState,
+    expectedProps: ExpectedTargetScopeProvenanceProps
+  ) {
+    const scopeProvenanceCandidates = branch.db.elements
+      .getAspects(IModelDb.rootSubjectId, ExternalSourceAspect.classFullName)
+      .filter(
+        (a) => (a as ExternalSourceAspect).identifier === master.db.iModelId
+      );
+    expect(scopeProvenanceCandidates).to.have.length(1);
+    const targetScopeProvenance =
+      scopeProvenanceCandidates[0].toJSON() as ExternalSourceAspectProps;
+    const targetScopeJsonProps = JSON.parse(
+      targetScopeProvenance.jsonProperties
+    ) as TargetScopeProvenanceJsonProps;
+
+    if (expectedProps.syncVersionId === undefined) {
+      const regex = new RegExp(`;${expectedProps.syncVersionIndex}$`);
```

Comment:
Q: Why this needs to be a regexp?

Reply:
Same question for reverseSyncVersion validation

Reply:
I added this as a helper function for some tests, in most cases when writing a test I don't know the changesetId so I will not pass it to this helper function. The regex allows us to assert that the index I passed is part of the sync /reversesyncversion.

Without regex I would always have to pass ID + index and then compare id;index to the actual id;index in the target ESA.

Maybe you're saying I could instead just do a endsWith comparison? (targetScopeProvenance.version?.endsWith(`;${syncVersionIndex}`)